### PR TITLE
use monotonically_increasing_id in zipWithUniqueId for 2.x

### PR DESCRIPTION
**Pull request by mengxr**
Both SPARK-9020 and SPARK-13473 were fixed in 2.x. This PR removes the workaround to get unique long IDs in 2.x.

### CODE REVIEW

File: src/main/spark-2.x/org/apache/spark/sql/SQLHelpers.scala [hunk 1]
```diff
@@ -28,17 +28,11 @@ object SQLHelpers {
...
-    val sqlContext = df.sqlContext
-    val schema = df.schema
-    val rdd = df.rdd.zipWithUniqueId().map { case (row, id) =>
-      Row(row, id)
-    }
-    val outputSchema = StructType(Seq(
-      StructField("row", schema, false), StructField("uniq_id", LongType, false)))
-    sqlContext.createDataFrame(rdd, outputSchema)
+    df.select(
+      struct(df.columns.map(c => df(c)) :_*).as("row"),
```
Comment (jkbradley) [inline src/main/spark-2.x/org/apache/spark/sql/SQLHelpers.scala]: I'm encouraging people to use ```nestAsCol```.

**Comment by jkbradley** [comment]
LGTM
I had one comment, if you don't mind fixing it.  Thanks!

**Comment by jkbradley** [comment]
Oh nevermind, that looks like an actual test failure

File: src/main/spark-2.x/org/apache/spark/sql/SQLHelpers.scala (hunk 1 above)
Comment (mengxr) [inline src/main/spark-2.x/org/apache/spark/sql/SQLHelpers.scala]: That method is not visible under this package.

**Comment by mengxr** [comment]
Closing this PR since `monotonically_increasing_id` still has problems. https://issues.apache.org/jira/browse/SPARK-14393

**Pull request closed without merging**